### DISCUSSION TITLE
make alert template more human readable

### DISCUSSION THREAD

**Discussion by yosiasz** [grafana/grafana#52619] [Alerting]
Greetings,

Are there plans to make templates more developer/human friendly

This whole squiggly mess is so ugly with typing in all of these {{{{{{

```
{{ define "mymessage" }}
  {{ if gt (len .Alerts.Firing) 0 }}
    {{ len .Alerts.Firing }} firing:
    {{ range .Alerts.Firing }} {{ template "myalert" .}} {{ end }}
  {{ end }}
  {{ if gt (len .Alerts.Resolved) 0 }}
    {{ len .Alerts.Resolved }} resolved:
    {{ range .Alerts.Resolved }} {{ template "myalert" .}} {{ end }}
  {{ end }}
{{ end }}
```

Why cant we eliminate all this junk and maybe even allow intellisense? Unless there is one via vscode if we choose to build our alerts via vscode with a special plugin? I think by far the the biggest frustration point and volume of questions for many grafana users must be alerts imho

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
